core/services/eth/errors.go

```diff
@@ -253,7 +253,18 @@ func ExtractRevertReasonFromRPCError(err error) (string, error) {
 	if err != nil {
 		return "", errors.Wrap(err, "unable to decode hex to bytes")
 	}
-	revertReasonBytes = bytes.Trim(revertReasonBytes, "\x00")
+
+	ln := len(revertReasonBytes)
```

Comment:
This is needed to handle a strings like "\x00\x00\x00 \x00\x00\x00upkeep not needed\x00 \x00". As you can see there is an empty space between "\x00" characters. If we do trim only one, there will be "\x00\x00upkeep not needed\x00" in the end. We need to trim all spaces and "null" characters.

Reply:
hallelujah thank you. This was driving me nuts. :raised_hands: